src/topology/instances/real.lean

```diff
@@ -108,6 +108,76 @@ noncompact_space_of_ne_bot $ by simp [at_top_ne_bot]
 
 end int
 
+
+namespace nat
+
+instance : has_dist ℕ := ⟨λ x y, dist (x : ℝ) y⟩
+
+theorem dist_eq (x y : ℕ) : dist x y = |x - y| := rfl
+
+lemma dist_coe_int (x y : ℕ) : dist (x : ℤ) (y : ℤ) = dist x y := rfl
+
+@[norm_cast, simp] theorem dist_cast_real (x y : ℕ) : dist (x : ℝ) y = dist x y := rfl
+
+@[norm_cast, simp] theorem dist_cast_rat (x y : ℕ) : dist (x : ℚ) y = dist x y :=
+by rw [← nat.dist_cast_real, ← rat.dist_cast]; congr' 1; norm_cast
+
+lemma pairwise_one_le_dist : pairwise (λ m n : ℕ, 1 ≤ dist m n) :=
+begin
+  intros m n hne,
+  rw ← dist_coe_int,
+  apply int.pairwise_one_le_dist,
+  exact_mod_cast hne
+end
+
+lemma uniform_embedding_coe_rat : uniform_embedding (coe : ℕ → ℚ) :=
+uniform_embedding_bot_of_pairwise_le_dist zero_lt_one $ by simpa using pairwise_one_le_dist
+
+lemma closed_embedding_coe_rat : closed_embedding (coe : ℕ → ℚ) :=
+closed_embedding_of_pairwise_le_dist zero_lt_one $ by simpa using pairwise_one_le_dist
+
+lemma uniform_embedding_coe_real : uniform_embedding (coe : ℕ → ℝ) :=
+uniform_embedding_bot_of_pairwise_le_dist zero_lt_one pairwise_one_le_dist
+
+lemma closed_embedding_coe_real : closed_embedding (coe : ℕ → ℝ) :=
+closed_embedding_of_pairwise_le_dist zero_lt_one pairwise_one_le_dist
+
+instance : metric_space ℕ := nat.uniform_embedding_coe_real.comap_metric_space _
+
+theorem preimage_ball (x : ℕ) (r : ℝ) : coe ⁻¹' (ball (x : ℝ) r) = ball x r := rfl
+
+theorem preimage_closed_ball (x : ℕ) (r : ℝ) :
+  coe ⁻¹' (closed_ball (x : ℝ) r) = closed_ball x r := rfl
+
+theorem closed_ball_eq_Icc (x : ℕ) (r : ℝ) :
+  closed_ball x r = Icc ⌈↑x - r⌉₊ ⌊↑x + r⌋₊ :=
+begin
+  rcases le_or_lt 0 r with hr|hr,
+  { rw [← preimage_closed_ball, real.closed_ball_eq_Icc, preimage_Icc],
+    exact add_nonneg (cast_nonneg x) hr },
+  { rw closed_ball_eq_empty.2 hr,
+    apply (Icc_eq_empty _).symm,
+    rw not_le,
+    calc ⌊(x : ℝ) + r⌋₊ ≤ ⌊(x : ℝ)⌋₊ : by { apply floor_mono, linarith }
+    ... < ⌈↑x - r⌉₊ : by { rw [floor_coe, nat.lt_ceil], linarith } }
+end
+
+instance : proper_space ℕ :=
+⟨ begin
+    intros x r,
+    rw closed_ball_eq_Icc,
+    exact (set.finite_Icc _ _).is_compact,
+  end ⟩
+
+@[simp] lemma cocompact_eq : cocompact ℕ = at_top :=
+by simp only [← comap_dist_right_at_top_eq_cocompact (0 : ℕ), dist_eq, sub_zero, cast_zero,
+  abs_cast, comap_coe_at_top]
```

Comment:
```suggestion
by rw [cocompact_eq_cofinite, cofinite_eq_at_top]
```

Reply:
In fact there is no such thing as `cocompact_eq_cofinite`, so I've reverted the change.